### PR TITLE
GNSS: Add support for constellation, measurement, clock and geofence

### PR DESCRIPTION
**Pull request by JianyuWang0623**
## Summary
1. Rename GPS to GNSS - 03f4ec7765ed8b3551f1a2bc8198c5007a542788
2. Add constellation for GNSS - 0f7860ac1160da0c559db9e2387b97d72dbedc26
3. Support GNSS Measurement & Clock - 49a96f16f9509a43c3d32f3712f87a6861a8ffed
4. Support GNSS Geofence - e738c34da2db4400096cf2f0fe9f8ad9db251beb
5. Let gnss geofence follow the existing naming rules - 47d9a8143ae09fa130be67300b02331d31979a43
6. Update GPSUTILS_MINMEA_LIB to GNSSUTILS_MINMEA_LIB - bbffbf0bb8eb8eb37856d937555f0d84682f83dc

## Impact
1. boards: CONFIG_GPSUTILS_MINMEA_LIB
2. drivers/sensors: SENSORS_GPS
3. include/nuttx
    1. sensors: struct gps_*
4. libs/libc: gpsutils
    1. gnssutils: libs/libc/gpsutils/minmea (download path)

## Testing
- Configuration
```
./tools/configure.sh -l sim:minmea
```
- Build
```
$ make -j$(nproc)
Downloading: gnssutils/minmea
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 23182    0 23182    0     0  11139      0 --:--:--  0:00:02 --:--:-- 88514
Archive:  db46128e73cee26d6a6eb0482dcba544ee1ea9f5.zip
db46128e73cee26d6a6eb0482dcba544ee1ea9f5
  inflating: gnssutils/minmea/c-cpp.yml  
  inflating: gnssutils/minmea/.gitignore  
  inflating: gnssutils/minmea/CMakeLists.txt  
  inflating: gnssutils/minmea/COPYING  
  inflating: gnssutils/minmea/README.md  
  inflating: gnssutils/minmea/minmea_compat_ti-rtos.h  
  inflating: gnssutils/minmea/minmea_compat_windows.h  
  inflating: gnssutils/minmea/example.c  
  inflating: gnssutils/minmea/minmea.c  
  inflating: gnssutils/minmea/minmea.h  
  inflating: gnssutils/minmea/tests.c  
CP:  /home/dramalife/upstream/github/nuttx/include/nuttx/config.h
CP:  /home/dramalife/upstream/github/nuttx/include/nuttx/fs/hostfs.h
LD:  nuttx
```

### CODE REVIEW

**Comment by Donny9** [comment]
This PR is related to https://github.com/apache/nuttx-apps/pull/2574
image

File: libs/libc/gpsutils/Kconfig [hunk 1]
```diff
@@ -12,3 +12,15 @@ config GPSUTILS_MINMEA_LIB
 		NOTE: This library depends on having some version of math.h
 		at include/nuttx.  There are some different ways to accomplish
 		this.  See the discussion in the top-level nuttx/README.txt file.
+
+if GPSUTILS_MINMEA_LIB
+
+config GPSUTILS_MINMEA_VERSION_4_10
```
Comment (xiaoxiang781216) [inline libs/libc/gpsutils/Kconfig]: no user, remove

File: include/nuttx/uorb.h [hunk 2]
```diff
@@ -678,4 +736,201 @@ struct sensor_ioctl_s
   char data[1];                /* The argument buf of ioctl */
 };
 
+/* This structure describes the information of the sensor device and
+ * requires the manufacturer to implement the device info function.
+ */
+
+struct sensor_device_info_s
```
Comment (xiaoxiang781216) [inline include/nuttx/uorb.h:781]: why add

Comment (JianyuWang0623) [reply include/nuttx/uorb.h:781]: It`s a mistake while processing merge conflict because of depends, fixed.

**Comment by anchao** [comment]
@xiaoxiang781216 @Donny9 please fix the build break:

image

**Pull request merged**